### PR TITLE
Add feature to use SingleShare to open WhatsApp chat screen from an App by passing whats mobile number of user.

### PR DESCRIPTION
**Pull request by shubhamdeol**
This is a feature, which solves this issue :

https://github.com/react-native-community/react-native-share/issues/531

### CODE REVIEW

File: android/src/main/java/cl/json/social/ShareIntent.java [hunk 1]
```diff
@@ -55,6 +55,18 @@ public void open(ReadableMap options) throws ActivityNotFoundException {
         if (ShareIntent.hasValidKey("message", options)) {
             message = options.getString("message");
         }
+        
+        String socialType = options.getString("social");
+
+        if(socialType.equals("whatsapp")) {
+            String whatsAppNumber = options.getString("whatsAppNumber");
+            if(!whatsAppNumber.isEmpty()) {
+            String chatAddress = whatsAppNumber + "@s.whatsapp.net";
```
Comment (MateusAndrade) [inline android/src/main/java/cl/json/social/ShareIntent.java]: you can add a identation here?

**Comment by MateusAndrade** [comment]
@shubhamdeol thanks for sending this PR. 😄 

Did you ind anything about how this can works on IOS?

File: android/src/main/java/cl/json/social/ShareIntent.java (hunk 1 above)
Comment (shubhamdeol) [reply android/src/main/java/cl/json/social/ShareIntent.java]: Sure.

**Comment by shubhamdeol** [comment]
@MateusAndrade  I have no idea about this thing for ios. I will research let you know if there is any way to do this with ios.

**Comment by MateusAndrade** [comment]
> @MateusAndrade I have no idea about this thing for ios. I will research let you know if there is any way to do this with ios.

Thanks. Adding this to both IOS/Android will be a great add for the project. Also, you can add to the example app this new feature? 😅

**Comment by MateusAndrade** [comment]
@shubhamdeol you can update your base branch with the master?

**Comment by shubhamdeol** [comment]
@MateusAndrade 
Thanks for letting me know. I was really confused about whether to update or not.
Can you please explain Is it really important to keep your pull request updated with base branch?

**Comment by MateusAndrade** [comment]
Sure @shubhamdeol ! You are just updating your base branch with the latest changes on the target branch. By doing this, you avoid some common mistakes like conflitcs, can track possible breaking changes, etc. 

I Will need to ask again to you to update your PR, because i Just merged some changes on master. :joy:

**Comment by shubhamdeol** [comment]
@MateusAndrade  lol, no worries.

**Comment by shubhamdeol** [comment]
@jgcmarins  yes, makes sense.

**Comment by MateusAndrade** [comment]
#554

**Pull request merged**